modules/foreman_job_template.py

```diff
@@ -0,0 +1,521 @@
+    if 'name' not in entity_dict:
+        if file_name:
+            entity_dict['name'] = os.path.splitext(
+                os.path.basename(file_name))[0]
+        else:
+            module.fail_json(
+                msg='No name specified and no filename to infer it.')
+
+    name = entity_dict['name']
+
+    affects_multiple = name == '*'
+    # sanitize user input, filter unuseful configuration combinations with 'name: *'
+    if affects_multiple:
+        if state == 'present_with_defaults':
+            module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
+        if state == 'absent':
+            if entity_dict.keys() != ['name']:
+                module.fail_json(msg="When deleting all job templates, there is no need to specify further parameters.")
+
+    try:
+        create_server(server_url, (username, password), verify_ssl)
+    except Exception as e:
+        module.fail_json(msg='Failed to connect to Foreman server: %s ' % e)
+
+    ping_server(module)
+
+    try:
+        if affects_multiple:
+            entities = find_entities(JobTemplate)
+        else:
+            entities = find_entities(JobTemplate, name=entity_dict['name'])
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
+
+    # Set Locations of job template
+    if 'locations' in entity_dict:
+        entity_dict['locations'] = find_entities_by_name(
+            Location, entity_dict['locations'], module)
+
+    # Set Organizations of job template
+    if 'organizations' in entity_dict:
+        entity_dict['organizations'] = find_entities_by_name(
+            Organization, entity_dict['organizations'], module)
+
+    # TemplateInputs need to be added as separate entities later
+    if 'template_inputs' in entity_dict:
+        template_input_list = entity_dict['template_inputs']
+
+    entity_dict = sanitize_entity_dict(entity_dict, name_map)
+
+    changed = False
+    if not affects_multiple:
+        if len(entities) == 0:
+            entity = None
+        else:
+            entity = entities[0]
+        changed, result = naildown_entity(
+            JobTemplate, entity_dict, entity, state, module)
+
+        if state == "present" or state == "present_with_defaults":
+
+            # Manage TemplateInputs here
+            for template_input_dict in template_input_list:
+                template_input_dict = template_input_dict.copy()
+
+                # assign template_input to a template
+                template_input_dict['template'] = result
+
+                ti_entity = find_template_input(module, str(template_input_dict['name']), result)
+
+                ti_changed, _ = naildown_entity(
+                    TemplateInput, template_input_dict, ti_entity, state, module)
+                changed |= ti_changed
+
+            # remove template inputs if they aren't present in template_input_list
+            found_tis = TemplateInput(template=result).search()
```

Comment:
This can bite you real hard if you have more then 20 inputs, because there is pagination in foreman. Better to use `find_entities` (https://github.com/theforeman/foreman-ansible-modules/blob/7a5c32671330223d0d0d556dc01d31309eb193a0/module_utils/ansible_nailgun_cement.py#L148).

Reply:
good catch!